src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -1864,14 +1864,18 @@ void SystemDictionary::add_nest_host_error(const constantPoolHandle& pool,
   {
     MutexLocker ml(Thread::current(), SystemDictionary_lock);
     ResolutionErrorEntry* entry = ResolutionErrorTable::find_entry(pool, which);
-    if (entry != nullptr && entry->nest_host_error() == nullptr) {
+    if (entry == nullptr) {
+      // Only add a new resolution error if one hasn't been found for this constant pool index. In this case,
+      // resolution succeeded but there's an error in this nest host.
```

Comment:
```suggestion
      // Only add a new entry to the resolution error table if one hasn't been found for this 
      // constant pool index. In this case resolution succeeded but there's an error in this nest host
      // that we use the table to record.
```

Reply:
I'm only re-stating, in a clearer way, what was previously there. I believe it is true.

Reply:
Yes, it is true, I didn't read your suggestion correctly.  It does clarify things and I shouldn't have called it a "resolution error" since that's not true.